apps/beeswax/src/beeswax/server/hive_server2_lib_tests.py

```diff
@@ -1041,3 +1041,44 @@ def test_call_session_close_idle_limit(self):
     finally:
       for f in finish:
         f()
+
+class TestHiveServerClientCompatible():
+
+  def setUp(self):
+    self.client = make_logged_in_client(username="test_hive_server2_lib", groupname="default", recreate=True, is_superuser=False)
+    self.user = User.objects.get(username="test_hive_server2_lib")
+    self.query_server = {
+        'principal': 'hue',
+        'server_name': 'hive',
+        'QUERY_TIMEOUT_S': 60,
+        'auth_username': 'hue',
+        'auth_password': 'hue',
+        'use_sasl': True,
+        'server_host': 'localhost',
+        'server_port': 10000,
+    }
+    self._client = HiveServerClient(self.query_server, self.user)
```

Comment:
I think we can Mock the whole `HiveServerClient` itself? That will remove the whole setUp() and also the patching needed?
Something like this?
```
client = Mock(
      query_server = Mock(),
      user = Mock(),
      get_tables_meta = Mock(return_value = [
        {'TABLE_NAME': 'sample_07', 'TABLE_TYPE': 'TABLE', 'REMARKS': None},
        {'TABLE_NAME': 'sample_08', 'TABLE_TYPE': 'TABLE', 'REMARKS': None},
        {'TABLE_NAME': 'web_logs', 'TABLE_TYPE': 'TABLE', 'REMARKS': None},
        {'TABLE_NAME': 'flights_13', 'TABLE_TYPE': 'TABLE', 'REMARKS': None},
        {'TABLE_NAME': 'flights', 'TABLE_TYPE': 'TABLE', 'REMARKS': None},
        {'TABLE_NAME': 'lights', 'TABLE_TYPE': 'TABLE', 'REMARKS': None}
      ])
    )
```

Reply:
:+1:

Reply:
Looks like we can do this way also because we are not using HiveServerClient really and if the client replaces the HiveServerClient then patching is also not needed in this way.
Thanks